### ISSUE TITLE
Action creators flow types

### ISSUE THREAD

**Issue by aloker** [erikras/redux-form#3278]
**bug report**

### What is the current behavior?
The action creators are currently exported with a type corresponding to the action they create.

lib/index.js.flow:
```
declare export var arrayInsert: ArrayInsertAction
declare export var arrayMove: ArrayMoveAction
declare export var arrayPop: ArrayPopAction
declare export var arrayPush: ArrayPushAction
declare export var arrayRemove: ArrayRemoveAction
declare export var arrayRemoveAll: ArrayRemoveAllAction
declare export var arrayShift: ArrayShiftAction
declare export var arraySplice: ArraySpliceAction
declare export var arraySwap: ArraySwapAction
declare export var arrayUnshift: ArrayUnshiftAction
declare export var autofill: AutofillAction
declare export var blur: BlurAction
declare export var change: ChangeAction
declare export var clearSubmitErrors: ClearSubmitErrorsAction
declare export var destroy: DestroyAction
declare export var focus: FocusAction
declare export var initialize: InitializeAction
declare export var registerField: RegisterFieldAction
declare export var reset: ResetAction
declare export var setSubmitFailed: SetSubmitFailedAction
declare export var setSubmitSucceeded: SetSubmitSucceededAction
declare export var startAsyncValidation: StartAsyncValidationAction
declare export var startSubmit: StartSubmitAction
declare export var stopAsyncValidation: StopAsyncValidationAction
declare export var stopSubmit: StopSubmitAction
declare export var submit: SubmitAction
declare export var touch: TouchAction
declare export var unregisterField: UnregisterFieldAction
declare export var untouch: UntouchAction
```
This causes type validation errors when using the creators.

E.g.
```
import { startSubmit } from 'redux-form'
...
yield put(startSubmit(formId))
...
```
```
Error: src/modules/forms/sagas/form-handler.js:8
  8:   yield put(startSubmit(formId))
                 ^^^^^^^^^^^^^^^^^^^ function call. Function cannot be called on any member of intersection type
  8:   yield put(startSubmit(formId))
                 ^^^^^^^^^^^ intersection
  Member 1:
                                       v
  179: export type StartSubmitAction = {
  180:   type: string,
  181:   meta: { form: string }
  182: } & Action
       ^ object type. See: node_modules/redux-form/lib/actions.types.js.flow:179
  Error:
    8:   yield put(startSubmit(formId))
                   ^^^^^^^^^^^^^^^^^^^ function call. Callable signature not found in
                                       v
  179: export type StartSubmitAction = {
  180:   type: string,
  181:   meta: { form: string }
  182: } & Action
       ^ object type. See: node_modules/redux-form/lib/actions.types.js.flow:179
  Member 2:
  182: } & Action
           ^^^^^^ Action. See: node_modules/redux-form/lib/actions.types.js.flow:182
  Error:
    8:   yield put(startSubmit(formId))
                   ^^^^^^^^^^^^^^^^^^^ function call. Callable signature not found in
  182: } & Action
           ^^^^^^ object type. See: node_modules/redux-form/lib/actions.types.js.flow:182

```

### What is the expected behavior?
The type of an action creator is a function RETURNING an action, not the type of the action itself. Therefore, this can be fixed by updating the types declarations:

```
declare export var arrayInsert: ArrayInsert
declare export var arrayMove: ArrayMove
declare export var arrayPop: ArrayPop
declare export var arrayPush: ArrayPush
declare export var arrayRemove: ArrayRemove
declare export var arrayRemoveAll: ArrayRemoveAll
declare export var arrayShift: ArrayShift
declare export var arraySplice: ArraySplice
declare export var arraySwap: ArraySwap
declare export var arrayUnshift: ArrayUnshift
declare export var autofill: Autofill
declare export var blur: Blur
declare export var change: Change
declare export var clearSubmitErrors: ClearSubmitErrors
declare export var destroy: Destroy
declare export var focus: Focus
declare export var initialize: Initialize
declare export var registerField: RegisterField
declare export var reset: Reset
declare export var setSubmitFailed: SetSubmitFailed
declare export var setSubmitSucceeded: SetSubmitSucceeded
declare export var startAsyncValidation: StartAsyncValidation
declare export var startSubmit: StartSubmit
declare export var stopAsyncValidation: StopAsyncValidation
declare export var stopSubmit: StopSubmit
declare export var submit: Submit
declare export var touch: Touch
declare export var unregisterField: UnregisterField
declare export var untouch: Untouch
```
Result:
```
import { startSubmit } from 'redux-form'
...
yield put(startSubmit(formId))
...
```
```
$ flow
No errors!
Done in 0.28s.
```

### What's your environment?
redux-form@7.0.3
flow-bin@0.51.1
Windows 10 x64

**Comment by mrharel** [comment]
There is still some errors for the `change` action creator:

```
app/shared/pages/contactPage/phoneNumbers.react.js:70
 70:             this.props.dispatch(change(FORM_NAME, "contact.secondaryPhone", null));
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function call
 70:             this.props.dispatch(change(FORM_NAME, "contact.secondaryPhone", null));
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ undefined (too few arguments, expected default/rest parameters). This type is incompatible with
116:     touch: boolean,
                ^^^^^^^ boolean. See: node_modules/redux-form/lib/actions.types.js.flow:116
```

The documentation of the method http://redux-form.com/7.0.3/docs/api/ActionCreators.md/ is 
```
change(form:String, field:String, value:any)
Saves the value to the field.
```
but there are two extra parameters: touch and persistentSubmitErrors
```
var change = function change(form, field, value, touch, persistentSubmitErrors) {
```
and if we don't pass them it throw this error.

**Comment by erikras** [comment]
Published as [`v7.0.4`](https://github.com/erikras/redux-form/releases/tag/v7.0.4).
